pkg/infra/metrics/metrics_test.go

```diff
@@ -0,0 +1,30 @@
+package metrics
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLabelNameSanitization(t *testing.T) {
+	testcases := []struct {
+		input    string
+		expected string
+		err      bool
+	}{
+		{input: "job", expected: "job"},
+		{input: "job._loal['", expected: "job_loal"},
```

Comment:
Does it work with spaces as well? Personally would replace spaces with underscore (_). WDYT?

```suggestion
		{input: "job._loal['", expected: "job_loal"},
		{input: "Data source'", expected: "Datasource"},
```

Reply:
Agree.